ts/components/CalendarsListContainer.tsx

```diff
@@ -74,81 +59,18 @@ const getCalendarsByAccount = (calendars: ReadonlyArray<Calendar>) => {
 /**
  * Allows the user to select one of the device available Calendars
  */
-class CalendarsListContainer extends React.PureComponent<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = INITIAL_STATE;
-  }
-
-  private renderSectionHeader = (info: { section: any }): React.ReactNode => (
-    <SectionHeaderComponent sectionHeader={info.section.title} />
-  );
-
-  private renderListItem = ({ item }: ListRenderItemInfo<Calendar>) => {
-    const { defaultCalendar } = this.props;
-    const isDefaultCalendar = defaultCalendar && item.id === defaultCalendar.id;
-    return (
-      <ListItemComponent
-        key={item.id}
-        title={convertLocalCalendarName(item.title)}
-        hideIcon={!isDefaultCalendar}
-        iconSize={12}
-        iconName={isDefaultCalendar ? "checkTickBig" : undefined}
-        onPress={() =>
-          isDefaultCalendar && this.props.onCalendarRemove
-            ? this.props.onCalendarRemove()
-            : this.props.onCalendarSelected(item)
-        }
-        accessible={true}
-        accessibilityRole={"radio"}
-        accessibilityLabel={`${convertLocalCalendarName(item.title)}, ${
-          isDefaultCalendar
-            ? I18n.t("global.accessibility.active")
-            : I18n.t("global.accessibility.inactive")
-        }`}
-      />
-    );
-  };
-
-  public render() {
-    const { calendarsByAccount } = this.state;
-    const { defaultCalendar } = this.props;
-
-    return (
-      <React.Fragment>
-        {pot.isError(calendarsByAccount) && (
-          <React.Fragment>
-            <Body>{mapResourceErrorToMessage(calendarsByAccount.error)}</Body>
-            <ButtonSolid
-              onPress={this.fetchCalendars}
-              label={I18n.t("global.buttons.retry")}
-              accessibilityLabel={I18n.t("global.buttons.retry")}
-            />
-          </React.Fragment>
-        )}
-        <View style={{ paddingHorizontal: customVariables.contentPadding }}>
-          {pot.isSome(calendarsByAccount) && (
-            <SectionList
-              extraData={{ defaultCalendar }}
-              sections={calendarsByAccount.value}
-              renderSectionHeader={this.renderSectionHeader}
-              renderItem={this.renderListItem}
-            />
-          )}
-          <EdgeBorderComponent />
-        </View>
-      </React.Fragment>
-    );
-  }
-
-  public componentDidMount() {
-    this.fetchCalendars();
-  }
-
-  private fetchCalendars = () => {
-    this.setState({
-      calendarsByAccount: pot.noneLoading
-    });
+const CalendarsListContainer = (props: Props) => {
+  const { onCalendarSelected, onCalendarsLoaded, onCalendarRemove } = props;
```

Comment:
```suggestion
const CalendarsListContainer = ( {
 onCalendarSelected, 
 onCalendarsLoaded,
 onCalendarRemove 
 }: Props) => {
```
What about destructuring the props here? What do you think?

Reply:
Yes, I normally prefer to destructure props as I had done, but that's ok too! I applied your suggestion